homeassistant/components/overkiz/water_heater_entities/hitachi_dhw.py

```diff
@@ -0,0 +1,95 @@
+"""Support for Hitachi DHW."""
+
+from typing import Any, cast
+
+from pyoverkiz.enums import OverkizCommand, OverkizCommandParam, OverkizState
+
+from homeassistant.components.water_heater import (
+    STATE_ECO,
+    STATE_HIGH_DEMAND,
+    STATE_OFF,
+    WaterHeaterEntity,
+    WaterHeaterEntityFeature,
+)
+from homeassistant.const import ATTR_TEMPERATURE, PRECISION_WHOLE, TEMP_CELSIUS
+
+from ..entity import OverkizEntity
+
+OVERKIZ_TO_OPERATION_MODE: dict[str, str] = {
+    OverkizCommandParam.STANDARD: STATE_ECO,
+    OverkizCommandParam.HIGH_DEMAND: STATE_HIGH_DEMAND,
+    OverkizCommandParam.STOP: STATE_OFF,
+}
+
+OPERATION_MODE_TO_OVERKIZ = {v: k for k, v in OVERKIZ_TO_OPERATION_MODE.items()}
+
+
+class HitachiDHW(OverkizEntity, WaterHeaterEntity):
+    """Representation of Hitachi DHW."""
+
+    _attr_min_temp = 30.0
+    _attr_max_temp = 70.0
+    _attr_precision = PRECISION_WHOLE
+
+    _attr_temperature_unit = TEMP_CELSIUS
+    _attr_supported_features = (
+        WaterHeaterEntityFeature.TARGET_TEMPERATURE
+        | WaterHeaterEntityFeature.OPERATION_MODE
+    )
+    _attr_operation_list = [*OPERATION_MODE_TO_OVERKIZ]
+
+    @property
+    def current_temperature(self) -> float:
+        """Return the current temperature."""
+        return cast(
+            float, self.executor.select_state(OverkizState.CORE_DHW_TEMPERATURE)
+        )
+
+    @property
+    def target_temperature(self) -> float:
+        """Return the temperature we try to reach."""
+        return cast(
+            float,
+            self.executor.select_state(
+                OverkizState.MODBUS_CONTROL_DHW_SETTING_TEMPERATURE
+            ),
+        )
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set new target temperature."""
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+        await self.executor.async_execute_command(
+            OverkizCommand.SET_CONTROL_DHW_SETTING_TEMPERATURE, cast(int, temperature)
```

Comment:
Why do we cast it to an int here, but on line 52 we cast it to a float?

And here we set the value, so we don't need to cast for type checking. If you want to change the value, you should make it:
```suggestion
        await self.executor.async_execute_command(
            OverkizCommand.SET_CONTROL_DHW_SETTING_TEMPERATURE, int(temperature)
```

Reply:
Here is the set of the temperature to the device..
The precision attr from the device says it supports only WHOLE temperature.
The previous code did that, I guess in order to send only valid value for this device.

Regarding the get of the temperature, casting it as float to respect what water entity is excptecting [in the doc](https://developers.home-assistant.io/docs/core/entity/water-heater/): a float

Reply:
We should have this validated by someone, I am not sure what the old component did. However it does not make sense to use `cast` here, since `self.executor.async_execute_command` does not care about the type for type checking.

You only should use `int()` or `float()` here if this is required by the command on the Overkiz side, but we need an user to test this. In the past sometimes it was a bit confusing where Overkiz wanted a float (even with precision whole).